homeassistant/util/async_.py

```diff
@@ -55,3 +56,26 @@ def run_callback() -> None:
 
     loop.call_soon_threadsafe(run_callback)
     return future
+
+
+def raise_in_loop() -> None:
+    """Raise if called inside the event loop."""
+    try:
+        get_running_loop()
+        in_loop = True
+    except RuntimeError:
+        in_loop = False
+
+    if in_loop:
+        raise RuntimeError("You cannot do I/O inside the event loop")
```

Comment:
This makes it a breaking change, right?  Integrations that do this will stop working.

Reply:
Based on our history of underestimating impact, maybe start with a warning?

Reply:
You guys sure love a backlash on reddit :)

Maybe make it optional ? Being the option an optout ?

Reply:
Doing I/O in the event loop breaks Home Assistant. It is causing a ton of issues. Optional is a bad idea.

Reply:
Maybe we should make the error more descriptive so the user knows what integration is causing the issue and what they should do to get it fixed?  Possibly have them update an issue listing the integration that needs to be fixed?

Reply:
I will update it to a warning and make it more descriptive. We will make it a hard error in a couple of releases.

Reply:
I don't know the full consequences of this action, but wondering if we can identify the integrations at fault that need to be fixed without breaking these integrations. Instead of breaking these integrations immediately, we can identify them and give some warning before blocking them. We have lived with many of these integrations for a while, I am sure we can survive for another release or two.

Reply:
@amelchio wrote:

> Based on our history of underestimating impact, maybe start with a warning

I should frame this and hang above my desk 😅

Reply:
Problem with the warning is that we don't have a stack trace, so we don't know the offending integration 😢

Reply:
let it crash an leave a configuration option to disable the patch for the time-being ? (opt out)

Reply:
Do we have the url they are calling ?

Reply:
@dgomes Throw by default with giving the option keep their offending integrations working until they can be fixed with something like `disable_asyncio_safety_checks = true` seems like the fastest path forward to getting integrations fixed as a warning in the log will likely be ignored since we can't expect most users to read the logs ..unless something is wrong.

Reply:
@bdraco sorry if I didn't express my self correctly, but that's what I'm proposing.

You don't want to break someones setup, and have them wait for an eventual fix. Opting out of the safety_check is very important.

Reply:
I think we are in violent agreement 👍